### DISCUSSION TITLE
limit number of cores used by `next build`

### DISCUSSION THREAD

**Discussion by aryankarim** [vercel/next.js#65983] [Help]
### Summary

Hello fellas,

I wanted to know if I could limit the number of physical CPU cores used by the `next build` command. I've looked at next and swc documentation, but there is no mention of this matter. 

Anyone has managed to do this?

**Comment by icyJoseph** [comment]
IIRC, you can use `experimental.cpus` in next.config.js:

- https://github.com/vercel/next.js/blob/canary/packages/next/src/server/config-shared.ts#L241

**Reply by aryankarim** [reply]
I've 16 cores.  I supplied 1 cpu like so, but weirdly 6 cores are heavily used for the build. Once I made it 16, all 16 threads were being used.

Am I correct or the margins are unnoticeable and these variables have no effect?

## 1 cpu
```
experimental: {
  workerThreads: false,
  cpus: 1,
},
```
![1 cpu](https://github.com/vercel/next.js/assets/58115075/b318136a-6b8c-47a7-a0a4-43dfa8492b57)


## 16 cpu
```
experimental: {
  workerThreads: false,
  cpus: 16,
},
```
![16 cpu](https://github.com/vercel/next.js/assets/58115075/3f8671ee-cbd7-4ba4-a457-dc5b3c070dd8)
